### ISSUE TITLE
Local variables are in syntax group juliaFunctionBlock

### ISSUE THREAD

**Issue by Donkeyfish87** [JuliaEditorSupport/julia-vim#172]
```julia
function foo(x, y)
    z = x + y
    return z
end
```

Both z, x and y belong to the syntax group juliaFunctionBlock, just as foo. 

I'd rather want two different groups, so it's possible to colorise function names without coloring all the variables used in the function too. Like the coloring here on github e.g.

Whether this is a bug, feature request, or silly is in the eye of the beholder I guess.

**Comment by Donkeyfish87** [comment]
I tried to solve this problem, but I'm struggling with vimscript.

The regex `syntax match juliaFunctionName "\v^\s*function \zs.*\ze\("` should match function name as far as I can see, and it does indeed work if I turn off the julia-vim package and put it in after/syntax/julia.vim.

However, when I include the julia-vim package, it fails to register. I've made sure to include `juliaFunctionName` in the @juliaExpressions as well to ensure that it should be treated as contained.

If I replace the regex with matching just a single word like Foo it works. If I disable julia-vim the regex works. In combination it doesn't.
